bitsandbytes/functional.py

```diff
@@ -2690,7 +2644,22 @@ def transform(A, to_order, from_order="row", out=None, transpose=False, state=No
     return out, new_state
 
 
-def spmm_coo(cooA, B, out=None):
+def spmm_coo(cooA: Union[COOSparseTensor, torch.Tensor], B: torch.Tensor, out: torch.Tensor = None):
+    if not isinstance(cooA, COOSparseTensor):
+        assert (
+            cooA.is_sparse and cooA.layout == torch.sparse_coo
+        ), "Tensor must be `COOSparseTensor or a PyTorch COO tensor."
```

Comment:
I wouldn't use an assert in runtime code, since they will be wiped out by `python -O`.
```suggestion
        if not (cooA.is_sparse and cooA.layout == torch.sparse_coo):
            raise ValueError("Tensor must be a `COOSparseTensor` or a PyTorch COO tensor.")
```

Reply:
We're planning to take a library-wide sweep through these separately. For now, I'm following the existing patterns, but it is on the backlog.